graph/graph.go

```diff
@@ -37,6 +37,22 @@ type Graph struct {
 	imageMutex imageMutex // protect images in driver.
 }
 
+type Image struct {
+	ID              string            `json:"id"`
+	Parent          string            `json:"parent,omitempty"`
+	Comment         string            `json:"comment,omitempty"`
+	Created         time.Time         `json:"created"`
+	Container       string            `json:"container,omitempty"`
+	ContainerConfig runconfig.Config  `json:"container_config,omitempty"`
+	DockerVersion   string            `json:"docker_version,omitempty"`
+	Author          string            `json:"author,omitempty"`
+	Config          *runconfig.Config `json:"config,omitempty"`
+	Architecture    string            `json:"architecture,omitempty"`
+	OS              string            `json:"os,omitempty"`
+	Size            int64
+	graph           Graph
```

Comment:
We just got done decoupling this object from the graph driver... this is pretty disappointing. This is an API object coupled to internal storage formats.